### PR TITLE
Refactor validations related to brave_publisher_id

### LINKED ISSUE #346: Provide means to retry setting of brave_publisher_id
As @ayumi [pointed out](https://github.com/brave-intl/publishers/pull/342#issuecomment-348647242), #342 introduces an edge case in which `brave_publisher_id` normalization / inspection fails and cannot be reattempted. We should ensure that `brave_publisher_id_error_code` can be cleared out so that users don't need to recreate an account to retry entering a domain.

### PR DESCRIPTION
**Pull request by dgeb**
This fixes a couple edge cases that were not handled well in #342:

* When domain normalization/inspection fails, it is now possible to reattempt submitting the domain (or an updated domain) in the contact info form.

* When a domain that matches an already verified domain has been entered, the error is now displayed properly to the user when JS is enabled.

Fixes #346 

/cc @ayumi 

Submitter Checklist:

- [X] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [X] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [X] Added/updated tests for this change (for new code or code which already has tests).
- [X] Tagged reviewers.

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

File: app/models/publisher.rb [hunk 1]
```diff
@@ -172,13 +181,19 @@ def generate_verification_token
     update_attribute(:verification_token, PublisherTokenRequester.new(publisher: self).perform)
   end
 
+  def clear_brave_publisher_id
```
Comment (ayumi) [inline app/models/publisher.rb]: I don't think this method is used

Comment (dgeb) [reply app/models/publisher.rb]: Ah good catch. This method was from an intermediate iteration and should have been abandoned. 👍

**Comment by dgeb** [comment]
Ok I'll include some more controller tests and try to wrap this up first thing tomorrow.

**Comment by dgeb** [comment]
Feeling better about this now. I've added controller tests for both xhr and standard paths to update the domain, showing a failure case followed by a success.

Please take a look when you're ready @ayumi   👀

**Pull request merged**